tests/test_eventually.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+#
+# License: BSD
+#   https://raw.githubusercontent.com/splintered-reality/py_trees/devel/LICENSE
+#
+##############################################################################
+# Imports
+##############################################################################
+
+import py_trees
+import py_trees.console as console
+import py_trees.tests
+import pytest
+
+##############################################################################
+# Eternal Guard
+##############################################################################
+
+
+def test_eventually() -> None:
+    console.banner("Eventually")
+
+    blackboard = py_trees.blackboard.Client()
+    blackboard.register_key(key="flag", access=py_trees.common.Access.WRITE)
+    blackboard.flag = False
+
+    set_flag_true = py_trees.behaviours.SetBlackboardVariable(
+        name="SetFlag", variable_name="flag", variable_value=True, overwrite=True
+    )
+    worker = py_trees.behaviours.TickCounter(
+        name="Counter-1", duration=1, completion_status=py_trees.common.Status.SUCCESS
+    )
+    parallel = py_trees.idioms.eventually(
+        name="Parallel", worker=worker, on_completion=set_flag_true
+    )
+    parallel.tick_once()
+    py_trees.tests.print_assert_banner()
+    py_trees.tests.print_assert_details(
+        text="BB Variable (flag)",
+        expected=False,
+        result=blackboard.flag,
+    )
+    print(py_trees.display.unicode_tree(root=parallel, show_status=True))
+    assert not blackboard.flag
+
+    parallel.tick_once()
+    py_trees.tests.print_assert_banner()
+    py_trees.tests.print_assert_details(
+        text="BB Variable (flag)",
+        expected=True,
+        result=blackboard.flag,
+    )
+    print(py_trees.display.unicode_tree(root=parallel, show_status=True))
+    assert blackboard.flag
+
+
+def test_eventually_swiss() -> None:
```

Comment:
I think the issue with `memory=False` doesn't arise in this test because the `on_failure` and `on_success` behaviors resolve in a single tick.